src/main/java/org/logstash/dissect/Dissector.java

```diff
@@ -0,0 +1,101 @@
+package org.logstash.dissect;
+
+import com.logstash.Event;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Dissector {
+    private final Pattern pattern = Pattern.compile("(.*?)%\\{(.*?)\\}");
+    private final List<Field> fields = new ArrayList<>();
+    private final List<Field> saveableFields = new ArrayList<>();
+    private final Map<Field, ValueRef> fieldValueRefMap = new HashMap<>();
+    private int initialOffset = 0;
+    private Field lastField;
+
+    public Dissector(String mapping) {
+        if (mapping.isEmpty()) return;
```

Comment:
This is a highly unusual thing to do in a constructor (in fact, I think it's the first time I've seen it ever). Do you consider it an error to get called with an empty mapping? If yes, then you should throw an `IllegalArgumentException`.